Server/building.ts

```diff
@@ -0,0 +1,191 @@
+import { Coordinate, Player, Unit, GameState, emptyPArr, PlayerArr, SocketEvent, Events } from "./types.js";
+import { compArr, concatEvents, dist, doubleIt } from "./utility.js";
+import { PlayerInfo } from "./player.js";
+import { withinRadius } from "../Client/functions.js";
+
+export { Building, BuildingStats};
+
+type BuildingStats  = {
+    name: string,
+    maxHealth: number;
+    damage: number; // Attack damage, 0 for doesn't attack?
+    splash: number; // splash radius, 0 for no splash/doesn't attack
+    range: number; // 0 for doesn't attack normally?
+    upkeep: number; // amount of energy required for upkeep
+    moneyGen: number; // Money generated at the start of each turn
+    energyGen: number; // Energy generated at the start of each turn
+    buildTime: number; 
+    size: number; // Buildings are assumed to be square
+    attributes: string[];
+}
+
+class Building {
+    public loc: Coordinate; // Coordinates of upper left tile
+    public stats: BuildingStats;
+    public owner: Coordinate; // [team, number] of player
```

Comment:
I have added a PlayerId type. I left the second element as number just in case we have teams with more than 2 people in the future. Too bad there's no "integer" type.